wsmaster/che-core-api-factory-gitlab/src/main/java/org/eclipse/che/api/factory/server/gitlab/GitlabOAuthTokenFetcher.java

```diff
@@ -103,6 +105,24 @@ public PersonalAccessToken fetchPersonalAccessToken(Subject cheSubject, String s
     }
   }
 
+  @Override
+  public Optional<Boolean> isValid(PersonalAccessToken personalAccessToken)
+      throws ScmCommunicationException, ScmUnauthorizedException {
+    if (gitlabApiClient == null
+        || !gitlabApiClient.isConnected(personalAccessToken.getScmProviderUrl())) {
+      LOG.debug(
+          "not a  valid url {} for current fetcher ", personalAccessToken.getScmProviderUrl());
+      return Optional.empty();
+    }
+    try {
+      GitlabOauthTokenInfo info = gitlabApiClient.tokenInfo(personalAccessToken.getToken());
+      return Optional.of(
+          Sets.difference(DEFAULT_TOKEN_SCOPE, Sets.newHashSet(info.getScope())).isEmpty());
```

Comment:
set equals ? same as in BitbucketServerPersonalAccessTokenFetcher .

Reply:
imo it's wrong. scopes must be rule `not-less-than` of what we require.